client.go

```diff
@@ -23,109 +23,108 @@ package orlop
 import (
 	"context"
 	"fmt"
+	"github.com/sirupsen/logrus"
+	"github.com/switch-bit/orlop/errors"
 	"github.com/switch-bit/orlop/log"
 	"github.com/switch-bit/orlop/version"
+	"go.opentelemetry.io/contrib/instrumentation/google.golang.org/grpc/otelgrpc"
 	"google.golang.org/grpc"
 	"google.golang.org/grpc/credentials"
 )
 
 // Connect creates a new client from configuration
+//
+// deprecated: use ConnectContext instead
 func Connect(cfg HasClientConfig, vault HasVaultConfig) (*grpc.ClientConn, error) {
 	return ConnectContext(context.Background(), cfg, vault)
 }
 
 // ConnectContext creates a new client from configuration
 func ConnectContext(ctx context.Context, cfg HasClientConfig, vault HasVaultConfig) (*grpc.ClientConn, error) {
-	var opts []grpc.DialOption
+	ctx, span := tracer.Start(ctx, "Connect")
+	defer span.End()
 
-	l := log.WithContext(ctx)
+	logger := log.FromContext(ctx)
+
+	var opts []grpc.DialOption
 
 	if len(cfg.GetURL()) == 0 {
-		l.Errorf("client: url required")
-		return nil, fmt.Errorf("client: url required")
+		err := errors.New("client: url required")
```

Comment:
Is there a way to get name of client here (ie bosun)? Not sure how to have more specific error message to pin down missing env var - @antonwinter and I were debugging based on this message yesterday